semantic/src/main/kotlin/tools/samt/semantic/SemanticChecker.kt

```diff
@@ -0,0 +1,28 @@
+package tools.samt.semantic
+
+import tools.samt.common.DiagnosticContext
+import tools.samt.parser.FileNode
+import tools.samt.semantic.checks.*
+
+internal interface SemanticCheck {
+    fun check(fileNode: FileNode)
+}
+
+class SemanticChecker private constructor(private val fileNode: FileNode, diagnostics: DiagnosticContext) {
```

Comment:
I still think we should move these checks into the SemanticModelBuilder as well.

Reply:
Yes will happen